blockball-bukkit-plugin/src/main/java/com/github/shynixn/blockball/bukkit/logic/business/listener/BallListener.kt

```diff
@@ -128,14 +136,22 @@ class BallListener @Inject constructor(
             val optBall = this.ballEntityService.findBallFromEntity(event.entity)
             if (optBall.isPresent) {
                 val ball = optBall.get()
-                ball.kickByEntity(event.damager)
+                val damager = event.damager
+
+                if (damager is Player && damager.isSneaking) {
+                    ball.grab(damager)
+                } else {
+                    ball.shootByPlayer(event.damager)
```

Comment:
shootByPlayer will throw an exception if event.damager is not a player. Please check it in the listener.

Reply:
Thanks for pointing that out